Description: Fix the indicator icon sometimes turning green when clicking on it.
Author: Felix Geyer <debfx-pkg@fobos.de>
Bug: https://bugs.launchpad.net/bugs/755537

--- plasma-widget-message-indicator-0.5.7.orig/src/message-indicator.h
+++ plasma-widget-message-indicator-0.5.7/src/message-indicator.h
@@ -61,6 +61,7 @@ private:
     Plasma::Svg* mSvg;
     QLabel* mNoIndicatorLabel;
     QWidget* mCurrentWidget;
+    Plasma::ItemStatus m_status;
 
     void initIcon();
     void initSourceModel();
--- plasma-widget-message-indicator-0.5.7.orig/src/message-indicator.cpp
+++ plasma-widget-message-indicator-0.5.7/src/message-indicator.cpp
@@ -45,6 +45,7 @@ MessageIndicator::MessageIndicator(QObje
 , mView(new ExpandedTreeView(mStack))
 , mSvg(new Plasma::Svg(this))
 , mNoIndicatorLabel(new QLabel(mStack))
+, m_status(Plasma::UnknownStatus)
 {
     setBackgroundHints(StandardBackground);
     setAspectRatioMode(Plasma::ConstrainedSquare);
@@ -191,7 +192,7 @@ void MessageIndicator::toolTipAboutToSho
     QSizeF oldSize = mSvg->size();
     // FIXME: Hardcoded size
     mSvg->resize(32, 32);
-    QString element = status() == Plasma::NeedsAttentionStatus ? NEW_STUFF_ICON : NO_NEW_STUFF_ICON;
+    QString element = m_status == Plasma::NeedsAttentionStatus ? NEW_STUFF_ICON : NO_NEW_STUFF_ICON;
     QPixmap pix = mSvg->pixmap(element);
     mSvg->resize(oldSize);
     toolTip.setImage(pix);
@@ -237,7 +238,7 @@ void MessageIndicator::paintInterface(QP
     int minSize = qMin(contentsRect.height(), contentsRect.width());
     QRect contentsSquare = QRect(contentsRect.x() + (contentsRect.width() - minSize) / 2, contentsRect.y() + (contentsRect.height() - minSize) / 2, minSize, minSize);
 
-    QString element = status() == Plasma::NeedsAttentionStatus ? NEW_STUFF_ICON : NO_NEW_STUFF_ICON;
+    QString element = m_status == Plasma::NeedsAttentionStatus ? NEW_STUFF_ICON : NO_NEW_STUFF_ICON;
     mSvg->paint(painter, contentsSquare, element);
 }
 
@@ -274,7 +275,6 @@ void MessageIndicator::slotDrawAttention
 
 void MessageIndicator::updateStatus()
 {
-    Plasma::ItemStatus status;
     if (mSourceModel && mSourceModel->rowCount() > 0) {
         // Check if one of the indicators want to draw attention
         QModelIndexList lst = mSourceModel->match(mSourceModel->index(0, 0),
@@ -282,18 +282,18 @@ void MessageIndicator::updateStatus()
                                                   QVariant(true),
                                                   1 /* hits */,
                                                   Qt::MatchExactly | Qt::MatchRecursive);
-        status = lst.isEmpty() ? Plasma::ActiveStatus : Plasma::NeedsAttentionStatus;
+        m_status = lst.isEmpty() ? Plasma::ActiveStatus : Plasma::NeedsAttentionStatus;
     } else {
-        status = Plasma::PassiveStatus;
+        m_status = Plasma::PassiveStatus;
     }
 
-    setStatus(status);
+    setStatus(m_status);
 
     // Update icon
     update();
 
     // Update views
-    if (status == Plasma::PassiveStatus) {
+    if (m_status == Plasma::PassiveStatus) {
         mView->hide();
         mNoIndicatorLabel->show();
         mCurrentWidget = mNoIndicatorLabel;
